src/util.js

```diff
@@ -107,3 +107,80 @@ export function format_memory_and_limit(usage, limit) {
         return _("");
     }
 }
+
+export function updateContainers() {
+    let newContainers = {};
+    let newContainersStats = {};
+    let newContainersMeta = [];
+    let inspectRet = [];
+    let containerStatsRet = [];
+    let countRun;
+    return new Promise((resolve, reject) => {
+        varlinkCall(PODMAN, "io.podman.ListContainers")
+                .then(reply => {
+                    newContainersMeta = reply.containers;
+                    if (!newContainersMeta) {
+                        console.log(newContainersMeta);
+                        resolve({newContainers: newContainers, newContainersStats: newContainersStats});
+                    }
+                    inspectRet = newContainersMeta.map((container) => {
+                        return new Promise((resolve, reject) => {
+                            varlinkCall(PODMAN, "io.podman.InspectContainer", {name: container.id})
+                                    .then(reply => {
+                                        resolve(JSON.parse(reply.container));
+                                    })
+                                    .catch(ex => reject(new Error("Failed to do InspectContainer call:", ex, JSON.stringify(ex))));
+                        });
+                    });
+                })
+                .then(function() {
+                    const containerCount = newContainersMeta.length;
+                    Promise.all(inspectRet)
+                            .then((inspectRet) => {
+                                let runEle = newContainersMeta.filter((ele) => {
+                                    return ele.status === "running";
+                                });
+                                inspectRet.map((inspectRet) => {
+                                    newContainers[inspectRet.ID] = inspectRet;
+                                });
+                                if (runEle.length === 0 && Object.keys(newContainers).length === containerCount) {
+                                    resolve({newContainers: newContainers, newContainersStats: newContainersStats});
+                                }
+                            })
+                            .catch(ex => console.error("Failed to do InspectContainer call:", ex, JSON.stringify(ex)));
+                })
+
+                .then(function() {
+                    containerStatsRet = newContainersMeta.filter((ele) => {
+                        return ele.status === "running";
+                    }).map((container) => {
```

Comment:
Please don't compute such values twice. I know, I just told you that the other one can go entirely, but if you ever need it twice, compute it at the top and assign it to a `const`.